src/modules/base/user-setting/user-setting.js

```diff
@@ -0,0 +1,171 @@
+(function() {
+    "use strict";
+
+    angular
+        .module("TendrlModule")
+        .controller("userSettingController", userSettingController);
+
+    /*@ngInject*/
+    function userSettingController($rootScope, $scope, $state, $stateParams, userStore, Notifications, AuthManager, loggedUser) {
+
+        var vm = this,
+            user,
+            currentUser,
+            isDataLoading,
+            typePassword,
+            confirmPassword,
+            currentState,
+            errorMsg;
+
+        vm.cancelModal = cancelModal;
+        vm.closeModal = closeModal;
+        vm.confirmModal = confirmModal;
+        vm.isDataLoading = true;
+        vm.typePassword = false;
+        vm.confirmPassword = false;
+        vm.toggleTypePassword = toggleTypePassword;
+        vm.toggleConfirmPassword = toggleConfirmPassword;
+        vm.currentState = $state.current.name;
+        vm.currentUser = AuthManager.getUserRole();
+        vm.errorMsg = "";
+
+        init();
+
+        function init() {
+            userStore.getUserInfo()
+                .then(function(data) {
+                    vm.isDataLoading = false;
+                    vm.user = data;
+                });
+        }
+
+        vm.modalHeader = {
+            "title": "My Settings",
+            "close": vm.closeModal
+        };
+
+        vm.modalFooter = [{
+            "name": "Cancel",
+            "type": "button",
+            "classname": "btn-default",
+            "onCall": vm.cancelModal
+        }, {
+            "name": "Save",
+            "type": "button",
+            "classname": "btn-primary",
+            "onCall": vm.confirmModal
+        }];
+
+        /**
+         * @name cancelModal
+         * @desc cancels the modal
+         * @memberOf deleteUserController                
+
+         */
+
+        function cancelModal() {
+            $state.go(vm.currentState);
+            $rootScope.$emit("modal.done", "cancel");
+        }
+
+        /**
+         * @name closeModal
+         * @desc close the modal
+         * @memberOf deleteUserController                
+
+         */
+        function closeModal() {
+            $rootScope.$emit("modal.done", "close");
+        }
+
+        /**
+         * @name next
+         * @desc takes to next step
+         * @memberOf deleteUserController                
+
+         */
+
+
+        function confirmModal() {
+            if (_validateUIFields()) {
+                vm.user.notification = vm.user.email_notifications;
+                userStore.editUser(vm.user)
+                    .then(function(data) {
+                        vm.closeModal();
+                        if (vm.currentState === "users") {
+                            userStore.getUserList()
+                                .then(function(data) {
+                                    if (data !== null) {
+                                        loggedUser.userList = data;
```

Comment:
@a2batic as we have dicussed, loggedUser.userList is not required now. You can directly broadcast data.